tests failing on 3.8

Two tests are failing [with python 3.8 on CI](https://travis-ci.org/samuelcolvin/pydantic/jobs/596131963):
* `test_initvars_post_init`
* `test_initvars_post_init_post_parse`

We should fix this since 3.8 is close to release. Any help would be much appreciated, I'll looked quickly but can't quite seem what's wrong.

traceback:

```
―――――――――――――――――――――― test_initvars_post_init_post_parse ――――――――――――――――――――――
    def test_initvars_post_init_post_parse():
        @pydantic.dataclasses.dataclass
>       class PathDataPostInitPostParse:
tests/test_dataclasses.py:504: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pydantic/dataclasses.py:137: in dataclass
    return wrap(_cls)
pydantic/dataclasses.py:132: in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen, config)
pydantic/dataclasses.py:84: in _process_class
    cls = dataclasses._process_class(_cls, init, repr, eq, order, unsafe_hash, frozen)  # type: ignore
/opt/python/3.8-dev/lib/python3.8/dataclasses.py:967: in _process_class
    str(inspect.signature(cls)).replace(' -> None', ''))
/opt/python/3.8-dev/lib/python3.8/inspect.py:3050: in __str__
    formatted = str(param)
/opt/python/3.8-dev/lib/python3.8/inspect.py:2568: in __str__
    formatannotation(self._annotation))
/opt/python/3.8-dev/lib/python3.8/inspect.py:1202: in formatannotation
    return repr(annotation)
/opt/python/3.8-dev/lib/python3.8/dataclasses.py:213: in __repr__
    return f'dataclasses.InitVar[{self.type.__name__}]'
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = typing.Union[pathlib.Path, NoneType], attr = '__name__'
    def __getattr__(self, attr):
        # We are careful for copy and pickle.
        # Also for simplicity we just don't relay all dunder names
        if '__origin__' in self.__dict__ and not _is_dunder(attr):
            return getattr(self.__origin__, attr)
>       raise AttributeError(attr)
E       AttributeError: __name__
/opt/python/3.8-dev/lib/python3.8/typing.py:757: AttributeError
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -51,6 +51,15 @@ jobs:
     - make test
     env:
     - 'CYTHON=yes'
+  - stage: test
+    python: 3.8
+    name: 'Cython: 3.8'
+    script:
+    - make build-cython-trace
+    - python -c "import sys, pydantic; print('compiled:', pydantic.compiled); sys.exit(0 if pydantic.compiled else 1)"
+    - make test
+    env:
+    - 'CYTHON=yes'
 
   - stage: test
     python: 3.6
@@ -68,6 +77,14 @@ jobs:
     - make test
     env:
     - 'DEPS=no'
+  - stage: test
+    python: 3.8
+    name: 'Without Deps 3.8'
+    script:
+      - pip uninstall -y cython email-validator typing-extensions devtools
+      - make test
+    env:
+      - 'DEPS=no'
 
   - stage: test
     python: 3.7

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,7 +7,7 @@ cache: pip
 python:
 - '3.6'
 - '3.7'
-- '3.8-dev'
+- '3.8'
 
 install:
 - make install
@@ -32,9 +32,6 @@ after_success:
 - bash <(curl -s https://codecov.io/bash) -e CYTHON,DEPS
 
 jobs:
-  allow_failures:
-  - python: '3.8-dev'
-
   include:
   - stage: test
     python: 3.6

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -9,6 +9,9 @@ strategy:
       python.version: '3.6'
     Python37:
       python.version: '3.7'
+# not supported as of 2019-10-16 :-(
+#    Python38:
+#      python.version: '3.8'
 
 steps:
 - task: UsePythonVersion@0

--- a/pydantic/main.py
+++ b/pydantic/main.py
@@ -144,7 +144,7 @@ UNTOUCHED_TYPES = FunctionType, property, type, classmethod, staticmethod
 
 
 class ModelMetaclass(ABCMeta):
-    @no_type_check
+    @no_type_check  # noqa C901
     def __new__(mcs, name, bases, namespace, **kwargs):  # noqa C901
         fields: Dict[str, ModelField] = {}
         config = BaseConfig

--- a/pydantic/main.py
+++ b/pydantic/main.py
@@ -144,8 +144,8 @@ UNTOUCHED_TYPES = FunctionType, property, type, classmethod, staticmethod
 
 
 class ModelMetaclass(ABCMeta):
-    @no_type_check  # noqa C901
-    def __new__(mcs, name, bases, namespace, **kwargs):
+    @no_type_check
+    def __new__(mcs, name, bases, namespace, **kwargs):  # noqa C901
         fields: Dict[str, ModelField] = {}
         config = BaseConfig
         validators: 'ValidatorListDict' = {}

--- a/tests/test_validators.py
+++ b/tests/test_validators.py
@@ -814,8 +814,8 @@ def test_root_validator_repeat():
             def root_validator_repeated(cls, values):
                 return values
 
-            @root_validator  # noqa: F811
-            def root_validator_repeated(cls, values):
+            @root_validator
+            def root_validator_repeated(cls, values):  # noqa: F811
                 return values
 
 
@@ -829,8 +829,8 @@ def test_root_validator_repeat2():
             def repeat_validator(cls, v):
                 return v
 
-            @root_validator(pre=True)  # noqa: F811
-            def repeat_validator(cls, values):
+            @root_validator(pre=True)
+            def repeat_validator(cls, values):  # noqa: F811
                 return values
 
 

--- a/tests/test_validators.py
+++ b/tests/test_validators.py
@@ -814,7 +814,7 @@ def test_root_validator_repeat():
             def root_validator_repeated(cls, values):
                 return values
 
-            @root_validator
+            @root_validator  # noqa: F811
             def root_validator_repeated(cls, values):  # noqa: F811
                 return values
 
@@ -829,7 +829,7 @@ def test_root_validator_repeat2():
             def repeat_validator(cls, v):
                 return v
 
-            @root_validator(pre=True)
+            @root_validator(pre=True)  # noqa: F811
             def repeat_validator(cls, values):  # noqa: F811
                 return values
 

